From cafcf64d3eeb446cffcd16f6cf793f9f10943b75 Mon Sep 17 00:00:00 2001
From: Jeff Sharkey <jsharkey@android.com>
Date: Wed, 7 Feb 2018 12:28:24 -0700
Subject: [PATCH 108/306] Let's not divide by zero.

Test: sure
Bug: 73060623
Change-Id: Ic9d60bb99e43a05c4131f642563d792de665e592
(cherry picked from commit f46dc156cfcb31da8714f29338614c2cd4fc61e7)
---
 .../com/android/server/net/NetworkPolicyManagerService.java     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/net/NetworkPolicyManagerService.java b/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
index a6f049e1d8d..a17dd122f76 100644
--- a/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
+++ b/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
@@ -1722,7 +1722,7 @@ public class NetworkPolicyManagerService extends INetworkPolicyManager.Stub {
                 final long totalBytes = getTotalBytes(
                         NetworkTemplate.buildTemplateMobileAll(state.subscriberId), start, end);
                 final long remainingBytes = limitBytes - totalBytes;
-                final long remainingDays = Math.min(1, (end - currentTimeMillis())
+                final long remainingDays = Math.max(1, (end - currentTimeMillis())
                         / TimeUnit.DAYS.toMillis(1));
                 if (remainingBytes > 0) {
                     quotaBytes = (remainingBytes / remainingDays) / 10;
-- 
2.17.1

